src/reducer.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * This action type will be dispatched when your history
+ * receives a location change.
+ */
+export const LOCATION_CHANGE = '@@router/LOCATION_CHANGE'
```

Comment:
I think we need more descriptive names so the difference between UPDATE_LOCATION and LOCATION_CHANGE is obvious. It particular we want to make it clear that only intercepting LOCATION_CHANGE is guaranteed to give you every navigation event. We might want to rename UPDATE_LOCATION to REQUEST_LOCATION_CHANGE, for example.